modules/minimint-mint/src/lib.rs

```diff
@@ -200,11 +201,13 @@ impl FederationModule for Mint {
     ) -> Result<InputMeta<'b>, Self::Error> {
         let meta = self.validate_input(interconnect, cache, input)?;
 
-        batch.append_from_iter(
-            input
-                .iter()
-                .map(|(_, coin)| BatchItem::insert_new(NonceKey(coin.0.clone()), ())),
-        );
+        batch.append_from_iter(input.iter().flat_map(|(amount, coin)| {
+            let key = NonceKey(coin.0.clone());
+            vec![
+                BatchItem::insert_new(key.clone(), ()),
+                BatchItem::insert_new(MintAuditItemKey::Redemption(key), amount),
+            ]
```

Comment:
nit: we could save an allocation by doing `chain(once(…), once(…))` here, alternatively, is `[T,2]: IntoIterator<IntoIter::Item=T>`?